### PR TITLE
ENH add DES pipeline packages imsupport, despyfits, despyastro, despymisc, and pixcorrect

### PR DESCRIPTION
**Pull request by beckermr**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/des-despyfits/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    - libload.patch
+    - libload_again.patch
+
+build:
+  skip: True  # [win]
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('c') }}
```
Comment (hmaarrfk) [inline recipes/des-despyfits/meta.yaml:21]:
```suggestion
    - {{ compiler('c') }}
    - m2-git   # [win]
```

try adding m2-git if patching is failing.

Comment (beckermr) [reply recipes/des-despyfits/meta.yaml:21]: ahhhhhh cool - I'll try that next! thank you!

Comment (beckermr) [reply recipes/des-despyfits/meta.yaml:21]: It turns out the madness of this PR is that one patch needs CRLF line endings and the other does not. 🤦 

Thanks for the note. We cannot build on windows so resolving this one.

Comment (hmaarrfk) [reply recipes/des-despyfits/meta.yaml:21]: i sometimes just copy my patched file over.

**Comment by beckermr** [comment]
ok @conda-forge/staged-recipes This one is ready for review!

A few notes

 - The windows failure should be ignored. The software only runs on osx or linux but one of the packages is noarch and so things are getting built on windows.
 - Many of the packages use non-standard layouts and build CDLL loaded into python using setuptools extensions. I added patches and other fixes to make things a bit more sane.
 - The imsupport package is header-only.

**Pull request merged**